### DISCUSSION TITLE
Keep a window's task in the same position when dragging the window into another monitor

### DISCUSSION THREAD

**Discussion by itaych** [lawand-dot-io/taskbar#104] [Feature requests]
**Describe the bug**
When moving a window between monitors, the button representing that window in the task bar shifts to the rightmost position.

**To Reproduce**
This sequence requires a multi monitor setup.
1. Open any two apps.
2. Select the app whose taskbar button is not the rightmost.
3. Drag it to the other monitor.

**Expected behavior**
The expected behavior is that nothing should happen, but the observed behavior is that the taskbar button representing the window you've moved will switch places to become the rightmost button.

**Desktop**
 - Taskbar version 1.2.5
 - OS: macOS 15.1 (24B83)
 - Chip: M3

**Comment by lawand-dot-io** [comment]
@itaych Hello. This is expected behaviour in case the "Displays have separate spaces" option in the system preferences in enabled. This option means that the window is effectively moved into a different space when it's dragged into another monitor, which removes it and adds it at the end of the list of windows in Taskbar. The same thing would happen if you are using one screen only and move a window into another space then back to the space it was at.

I understand how this could be confusing so I will consider a solution that keeps it shown at the same spot, if that was technically not too complex.

As a temporary solution, consider disabling the "Displays have separate spaces" option in the system preferences (in the Desktop & Dock section).

Let me know if you have further questions. Thanks

**Reply by itaych** [reply]
This workaround works, thank you.

**Reply by lawand-dot-io** [reply]
Nice

**Comment by itaych** [comment]
It took me some time to realize this, but your suggested workaround causes an unwanted side effect, in which the menu bar at the top of the screen no longer appears on all monitors. So it's not really a good solution. Thanks.

**Reply by lawand-dot-io** [reply]
I thought about this fix that you're asking for and I can do it. The next version of Taskbar will no longer change the position if the window is moved into another monitor, even if the "displays have separate spaces" option isn't turned on 👍

**Comment by lawand-dot-io** [comment]
Duplicate https://github.com/lawand-dot-io/taskbar/issues/128

**Comment by lawand-dot-io** [comment]
@itaych how is this working for you now that multi-monitor support is added? I think I can close this request now

**Reply by itaych** [reply]
Hey @lawand-dot-io, I appreciate the work you've done on this but my work computer is not a Mac anymore so I'm sorry to say I can't verify your fix at this point. Thank you.

**Reply by lawand-dot-io** [reply]
@itaych I understand. Thanks and have a nice day

**Comment by rrisk93** [comment]
Unfortunately, same thing happened to me....

MacBook Pro M3 
Sequoia 15.4
Taskbar 1.4.5
"Display have separate Spaces" option ON

**Reply by lawand-dot-io** [reply]
Hello. Thanks for the info. This is the expected behaviour for the current version of Taskbar. Whenever a window is moved into another space or moved into another monitor with "Display have separate Spaces" being turned on, the window order will be set at the highest order and it will be moved into the end of the list at the right. I have in mind to either make this configurable or solve the issue somehow in a future version. Thanks for understanding.
